hardswish: add backards pass test

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36420 hardswish: add backards pass test**
* #36351 hardsigmoid: add cuda kernels

Summary:

Adds a unit test for hardswish backward pass

Test Plan:

Unit test passes on cpu and cuda

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20994100](https://our.internmc.facebook.com/intern/diff/D20994100)

vkuzo committed:
hardswish: add backards pass test

Summary:

Adds a unit test for hardswish backward pass

Test Plan:

Unit test passes on cpu and cuda

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/test_nn.py:
```diff
@@ -10260,6 +10260,12 @@ def test_hardsigmoid_grad(self, device):
+    def test_hardswish_grad(self, device):
+        inputs = (torch.randn(4, 16, 16, device=device) - 0.5) * 10
+        inputs.requires_grad = True
+        self.assertTrue(gradcheck(F.hardswish, (inputs,)))
```
How does this work? Where is the reference output provided here?

vkuzo replied:
this is a nice util function, it checks that the gradient computed analytically by perturbing the input matches the gradient defined by the backwards pass.  It's used to test other backwards passes in this file.

https://github.com/pytorch/pytorch/blob/73a36a47a5d371f34c750b442a4258fe5fc97504/torch/autograd/gradcheck.py#L198 has more context

vkuzo committed:
Update on "hardswish: add backards pass test"


Summary:

Adds a unit test for hardswish backward pass

Test Plan:

Unit test passes on cpu and cuda

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20994100](https://our.internmc.facebook.com/intern/diff/D20994100)

[ghstack-poisoned]

vkuzo committed:
Update on "hardswish: add backards pass test"


Summary:

Adds a unit test for hardswish backward pass

Test Plan:

Unit test passes on cpu and cuda

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20994100](https://our.internmc.facebook.com/intern/diff/D20994100)

[ghstack-poisoned]
